client/Windows/wf_defaults.c

```diff
@@ -0,0 +1,112 @@
+#include <Windows.h>
+#include <wincred.h>
+#include <stdio.h>
+#include <malloc.h>
+
+#pragma warning(disable : 4706 4094) // assignment within conditional expression
+
+#define FreeRDP_ServerHostname (20)
+#define FreeRDP_Username (21)
+#define FreeRDP_Password (22)
+#define FreeRDP_GatewayHostname (1986)
+#define FreeRDP_GatewayUsername (1987)
+#define FreeRDP_GatewayPassword (1988)
+
+typedef struct rdp_settings rdpSettings;
+
+EXTERN_C_START
+
+const char* freerdp_settings_get_string(const rdpSettings* settings, size_t id);
+BOOL freerdp_settings_set_string(rdpSettings* settings, size_t id, const char* param);
+
+EXTERN_C_END
+
+PCWSTR ValidateString(const BYTE* pb, ULONG cb)
+{
+	return pb &&                                       // present ?
+	               cb &&                               // size not 0 ?
+	               !(cb & (sizeof(WCHAR) - 1)) &&      // cb == n * sizeof(WCHAR) ?
+	               !*(WCHAR*)(pb + cb - sizeof(WCHAR)) // 0 terminated ?
+	           ? (PCWSTR)pb
+	           : 0;
+}
+
+void AddDefaultSettings_I(rdpSettings* settings, size_t idHostname, size_t idUsername,
+                          size_t idPassword)
+{
+	PCSTR ServerHostname = freerdp_settings_get_string(settings, idHostname);
+
+	if (!ServerHostname)
+	{
+		return;
+	}
+
+	BOOL bExistUserName = 0 != freerdp_settings_get_string(settings, idUsername);
```

Comment:
No variable declaration in the middle of the block

Reply:
@hardening no longer strict here, changed the `C` standard recently ;)

Reply:
nice to know !